crates/sui-core/src/authority.rs

```diff
@@ -984,26 +1050,29 @@ impl AuthorityState {
                         },
                     )) = item
                     {
-                        if let Err(e) = self.process_one_tx(seq, &digest).await {
+                        self.metrics.post_processing_latest_seq_seen.set(seq as i64);
+                        self.metrics
+                            .post_processing_total_tx_sent_to_post_processing
+                            .inc();
+                        if let Err(e) = self.post_process_one_tx(seq, &digest).await {
                             warn!(?digest, "Couldn't process tx: {}", e);
                         }
                     }
                 }
-
-                // For both the error cases, we exit the loop which ends this task.
-                // TODO: Automatically restart the task, which in combination with the todo above,
-                // will process any skipped txes and then begin listening for new ones.
                 Err(RecvError::Closed) => {
-                    // The service closed the channel.
-                    error!("run_tx_post_processing_process receiver channel closed");
+                    // This shall not happen.
+                    error!("run_tx_post_processing_process receiver channel closed. This shall not happen.");
```

Comment:
I'm not sure I understand the idea behind having a log line that, if printed, claims that it couldn't have...

Reply:
Perhaps change the comment to explain why this shouldn't happen.

Reply:
sorry, missed this comment! will submit another PR

Reply:
https://github.com/MystenLabs/sui/pull/5128